crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -424,8 +427,13 @@ async fn c8y_mapper_syncs_pending_alarms_on_startup() {
     //     .await
     //     .unwrap();
 
+    // wait for session to get closed properly.
+    // Note: Need to find a better way to clean the session in the broker
+
+    tokio::time::sleep(Duration::from_secs(6)).await;
```

Comment:
I would prefer to fix the bug rather than just manage to have the test green.

That said:
* Update the keep alive to the minimum (1 s) and reduce to the sleep accordingly (2s). Having tests that wait for ages is a pain.
* I prefer this version waiting for the connection to be ended by the server (without clearing the session), compared to the previous proposal to clear the session, client side (hence throwing away all the pending messages not just the delivery status of the retained messages).
* And finally, the comment is still not telling what you do and why. I.e waiting that the keep-alive timeout elapses and that the server reset the sent status of the retained messages.

Reply:
> I would prefer to fix the bug rather than just manage to have the test green.

I agree with you, if it's going to take longer to fix this in the mapper code, then we may consider this as a workaround.

> 
> That said:
> 
> * Update the keep alive to the minimum (1 s) and reduce to the sleep accordingly (2s). Having tests that wait for ages is a pain.

I tried with the smaller keep-alive value (1s) and wait with 2s but, it does not work, got below error message
c8y_mapper_syncs_pending_alarms_on_startup' panicked at 'Keep alives should be >= 5  secs', .cargo/registry/src/github.com-1ecc6299db9ec823/rumqttc-0.10.0/src/lib.rs:390:


> * I prefer this version waiting for the connection to be ended by the server (without clearing the session), compared to the previous proposal to clear the session, client side (hence throwing away all the pending messages not just the delivery status of the retained messages).
> * And finally, the comment is still not telling what you do and why. I.e waiting that the keep-alive timeout elapses and that the server reset the sent status of the retained messages.

Updated the comment.